expression/builtin_string.go

```diff
@@ -1470,56 +1473,225 @@ func trimRight(str, remstr string) string {
 	}
 }
 
-type rpadFunctionClass struct {
+func getFlen4LpadAndRpad(sc *variable.StatementContext, arg Expression) (int, error) {
+	if constant, ok := arg.(*Constant); ok {
+		length, isNull, err := constant.EvalInt(nil, sc)
+		if isNull || err != nil || length > mysql.MaxBlobWidth {
+			return mysql.MaxBlobWidth, errors.Trace(err)
```

Comment:
If err is not nil, we should not raise an error, because this function may be not executed in fact.